src/core/control/jobs/RenderJob.h

```diff
@@ -45,6 +46,8 @@ class RenderJob: public Job {
 
     void rerenderRectangle(xoj::util::Rectangle<double> const& rect);
 
+    void renderToBuffer(xoj::util::CairoSurfaceWrapper& buffer) const;
+
```

Comment:
Pass smart_ptrs by value or its content by (const)-reference. In this case, I would pass the `cairo_surface_t*` itself.

Reply:
What the rational behind this? If ref/unref can be costful, then passing them by reference seems a sound idea.

Reply:
The rational behind this is, that the function actually does nothing with the smart_ptr. It does something with the object itself. Therefore it is cleaner to pass the object* instead of the smart ptr. Only if the function want to modify the smart_ptr itself, it is required to pass it.